diff -Naru slurm-docker-cluster/.env slurm-docker-cluster.centos9/.env
--- slurm-docker-cluster/.env	2025-10-16 16:48:51
+++ slurm-docker-cluster.centos9/.env	2025-10-16 17:13:35
@@ -3,4 +3,4 @@
 
 # Image version used to tag the container at build time (Typically matches the
 # Slurm tag in semantic version form)
-IMAGE_TAG=25.05.3-dev-tmp
+IMAGE_TAG=25.05.3-centos9-dev
diff -Naru slurm-docker-cluster/Dockerfile slurm-docker-cluster.centos9/Dockerfile
--- slurm-docker-cluster/Dockerfile	2025-10-16 16:36:57
+++ slurm-docker-cluster.centos9/Dockerfile	2025-10-16 17:13:57
@@ -1,4 +1,4 @@
-FROM rockylinux:9
+FROM quay.io/centos/centos:stream9
 
 LABEL org.opencontainers.image.source="https://github.com/giovtorres/slurm-docker-cluster" \
       org.opencontainers.image.title="slurm-docker-cluster" \
